elfi/methods/methods.py

```diff
@@ -635,40 +635,59 @@ class BOLFI(InferenceMethod):
 
     """
 
-    def __init__(self, model, batch_size=1, discrepancy=None, bounds=None, **kwargs):
-        """
+    def get_posterior(self, threshold=None):
```

Comment:
Perhaps this could be renamed to something that better reflects the fact that we do quite a bit of work here to infer the posterior. E.g. `infer_posterior` ?